### DISCUSSION TITLE
Google Photorealistic 3D Tiles rendered with VSG and vsgCs

### DISCUSSION THREAD

**Discussion by timoore** [vsg-dev/VulkanSceneGraph#808] [General]
Google recently announced [Photorealistic 3D Tiles](https://cloud.google.com/blog/products/maps-platform/create-immersive-3d-map-experiences-photorealistic-3d-tiles) which is a very high resolution globe-scale dataset. The data is in the 3D Tiles format rendered by vsgCs, so (after a bug-fix) this new data works nicely:
![Screenshot from 2023-05-13 03-13-36](https://github.com/vsg-dev/VulkanSceneGraph/assets/133121/11b28f65-c5d0-4e04-955c-a9836b65bc3f)
![Screenshot from 2023-05-13 03-22-57](https://github.com/vsg-dev/VulkanSceneGraph/assets/133121/acdbc596-a918-4f56-9142-ed33cb83fad1)

**Comment by robertosfield** [comment]
When I saw the announcement from Cesium/Google my first thought could we try out the data and you've answered it in the best way possible.  What is the process for getting it running?

**Reply by timoore** [reply]
> When I saw the announcement from Cesium/Google my first thought could we try out the data and you've answered it in the best way possible. What is the process for getting it running?

* update to latest [vsgCs](https://github.com/timoore/vsgCs)
* Generate a Google Maps API key with access to the Map Tiles API. I've never used the Google Cloud platform before, but I followed [these directions](https://developers.google.com/maps/documentation/embed/cloud-setup) with a bit of hassle
* Use this world file for vsgCs:

```
{ "Type": "World",
  "tilesets": [
      { "Type": "Tileset",
        "tilesetUrl":"https://tile.googleapis.com/v1/3dtiles/root.json?key=MYKEY"
      }
  ]
}
```
substituting your Google Maps API key for MYKEY.

**Reply by robertosfield** [reply]
Thanks for the instructions, I won't have time right away to follow up but will certainly have a tinker when I can.

Have you come across any official support for open source/open communities w.r.t to access to the data?

**Reply by robertosfield** [reply]
I have had bash at creating an Google Maps Platform account and have a key but when I use it I just get 403:

$ bin/worldviewer --world-file tests/google.json 
[2023-05-17 17:26:48.409] [info] [GltfConverters.cpp:15] Registering magic header glTF
[2023-05-17 17:26:48.410] [info] [GltfConverters.cpp:15] Registering magic header b3dm
[2023-05-17 17:26:48.410] [info] [GltfConverters.cpp:15] Registering magic header cmpt
[2023-05-17 17:26:48.410] [info] [GltfConverters.cpp:15] Registering magic header pnts
[2023-05-17 17:26:48.410] [info] [GltfConverters.cpp:22] Registering file extension .gltf
[2023-05-17 17:26:48.410] [info] [GltfConverters.cpp:22] Registering file extension .glb
Warning: Received status code 403 for tileset https://tile.googleapis.com/v1/3dtiles/root.json?key=

I use my key provided by the Maps API key but don't include it above,  I'm a bit lost about exactly what is what though.

**Comment by timoore** [comment]
Hmm, works for me still.
Are you sure you have a Map Tiles API key?
Also make sure you are up-to-date with Cesium Native, although that  should
not be a problem.

Tim

**Comment by robertosfield** [comment]
I updated vsgCs but not Cesium Native.  I'll do that tomorrow when I am back at my dev system

**Reply by robertosfield** [reply]
I have updated cesium-native, built and installed, then rebuilt and install vsgCs but I still get the same Received status code 403.

Do I have to do anything more within my google map platform page to enable it?

**Comment by timoore** [comment]
I don't know the ins and outs of Google Cloud and the Maps API. One
possibility is that the Map Tiles API needs to be enabled specifically,
even if you have a Maps API key. This search might provide some clues:

https://community.cesium.com/search?q=403%20order%3Alatest

Tim

**Comment by gwaldron** [comment]
Excellent @timoore :)
